godot_project/project_workspace/workspace_context/structure_context/selection_db/atom_selection/atom_selection.gd

```diff
@@ -233,7 +233,12 @@ func can_grow_selection() -> bool:
 
 
 func grow_selection() -> AtomSelectionResult:
-	var atoms_to_select: PackedInt32Array = PackedInt32Array()
+	var atoms_to_select_dict: Dictionary = {}
+	# In the list of atoms to select we include already selected atoms,
+	# this helps to automatically expand selections of bonds in some corner cases
+	for atom_id: int in get_atoms_selection():
+		atoms_to_select_dict[atom_id] = true
+	
```

Comment:
I'm wondering if it would not be more straightforward if we would replace logic of this function with something like (pseudocode):
```
var atom_selection = get_atoms_selection()
for atom_id in atom_selection:
   var related_bonds = atom_get_bonds(atom_id)
   for bond_id in related_bonds:
      (...)
```